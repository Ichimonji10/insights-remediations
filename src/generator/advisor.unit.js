'use strict';

const { request, reqId } = require('../test');

test('generates a rule-based playbook', () => {
    const data = {
        issues: [{
            id: 'advisor:bond_config_issue|BOND_CONFIG_ISSUE',
            systems: ['68799a02-8be9-11e8-9eb6-529269fb1459']
        }]
    };

    return request
    .post('/v1/playbook')
    .send(data)
    .expect(200)
    .then(res => expect(res.text).toMatchSnapshot());
});

test('400s on unknown issue id', () => {
    const {id, header} = reqId();

    return request
    .post('/v1/playbook')
    .set(header)
    .send({
        issues: [{
            id: 'advisor:nonExistentId',
            systems: ['68799a02-8be9-11e8-9eb6-529269fb1459']
        }]
    })
    .expect(400)
    .then(({ body }) => {
        body.errors.should.eql([{
            id,
            status: 400,
            code: 'UNKNOWN_ISSUE',
            title: 'Unknown issue identifier "advisor:nonExistentId"'
        }]);
    });
});

test('400s on unsupported issue', () => {
    const {id, header} = reqId();

    return request
    .post('/v1/playbook')
    .set(header)
    .send({
        issues: [{
            id: 'advisor:alias_interface_invalid|ALIAS_INTERFACE_INVALID',
            systems: ['68799a02-8be9-11e8-9eb6-529269fb1459']
        }]
    })
    .expect(400)
    .then(({ body }) => {
        body.errors.should.eql([{
            id,
            status: 400,
            code: 'UNSUPPORTED_ISSUE',
            title: 'Issue "advisor:alias_interface_invalid|ALIAS_INTERFACE_INVALID" does not have Ansible support'
        }]);
    });
});

test('puts quotes around hosts list', async () => {
    const {text} = await request
    .post('/v1/playbook')
    .send({
        issues: [{
            id: 'advisor:bond_config_issue|NO_QUOTES',
            systems: ['68799a02-8be9-11e8-9eb6-529269fb1459']
        }]
    }).expect(200);

    expect(text).toMatchSnapshot();
});
