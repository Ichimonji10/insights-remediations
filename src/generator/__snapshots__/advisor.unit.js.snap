// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates a rule-based playbook 1`] = `
"---
- name: run insights to obtain latest report info
  hosts: \\"a8799a02-8be9-11e8-9eb6-529269fb1459.example.com\\"
  become: True
  tasks:
    - name: determine insights version
      shell: 'redhat-access-insights --version'
      changed_when: false
      register: insights_version

    - when: insights_version.stdout[0:2] != '1.'
      block:
        - name: obtaining insights report
          shell: 'redhat-access-insights --to-json --quiet'
          register: insights_result
          changed_when: false
          check_mode: false
        - name: register insights report as fact for use by other plays
          set_fact: insights_report={{ insights_result.stdout }}

    - when: insights_version.stdout[0:2] == '1.'
      block:
        - name: obtaining insights report (legacy client)
          shell: 'redhat-access-insights --verbose | grep \\"Upload status: 201 Created\\" | grep -o \\"{.*}\\"'
          register: insights_result_legacy
          changed_when: false
          check_mode: false
        - name: register insights report as fact for use by other plays (legacy client)
          set_fact: insights_report={{ insights_result_legacy.stdout }}

# Unexpected bonding behavior with incorrect syntax in bond configuration files
# Identifier: (advisor:bond_config_issue|BOND_CONFIG_ISSUE,fix)
# Version: unknown
- name: Correct Bonding Config Items
  hosts: \\"a8799a02-8be9-11e8-9eb6-529269fb1459.example.com\\"
  become: true
  vars:
    pydata: \\"{{ insights_report.details['bond_config_issue|BOND_CONFIG_ISSUE'] }}\\"
  tasks:

    - when: 
        - insights_report.details['bond_config_issue|BOND_CONFIG_ISSUE'] is defined
        - item.value == 2
      name: Add quotes around bonding options
      lineinfile:
        dest: \\"/etc/sysconfig/network-scripts/ifcfg-{{ item.key }}\\"
        regexp: '(^\\\\s*BONDING_OPTS=)(.*)'
        backrefs: yes
        line: '\\\\1\\"\\\\2\\"'
      with_dict: \\"{{ pydata.interface_issue_dict }}\\"

    - when:
        - insights_report.details['bond_config_issue|BOND_CONFIG_ISSUE'] is defined
        - item.value == 1
      name: lowercase yes in Slave option
      lineinfile:
        dest: \\"/etc/sysconfig/network-scripts/ifcfg-{{ item.key }}\\"
        regexp: '(^\\\\s*SLAVE=)(\\"*YES\\"*)'
        backrefs: yes
        line: '\\\\1yes'
      with_dict: \\"{{ pydata.interface_issue_dict }}\\"


- name: run insights
  hosts: \\"a8799a02-8be9-11e8-9eb6-529269fb1459.example.com\\"
  become: True
  gather_facts: False
  tasks:
    - name: run insights
      command: redhat-access-insights
      changed_when: false"
`;
